spmd/__init__.py

```diff
@@ -1,103 +1,86 @@
 # Copyright (c) Meta Platforms, Inc. and affiliates
-from typing import Sequence, Optional, cast
+from typing import Optional, Callable, Tuple, Union
 import torch
 import torch.nn as nn
-from spmd.tensor import DTensor, Placement, Shard, Replicate
+from spmd.tensor import distribute_tensor
+from spmd.tensor.api import DTensor
 from spmd.tensor.device_mesh import get_global_device_mesh, DeviceMesh
+from spmd.tensor.placement_types import Replicate, Shard
 
-torch.__future__.set_overwrite_module_params_on_conversion(True)
 
-
-def distribute_tensor(
-    tensor: torch.Tensor,
+def distribute_module(
+    module: nn.Module,
     device_mesh: Optional[DeviceMesh] = None,
-    placements: Optional[Sequence[Placement]] = None,
-) -> DTensor:
+    partition_fn: Optional[Callable[[str, nn.Module], None]] = None,
+    input_fn: Optional[Callable[..., None]] = None,
+    output_fn: Optional[Callable[..., None]] = None,
+) -> nn.Module:
```

Comment:
Another question is, I believe we are doing in-place modification to `module` then why do we still return this module? It can be confusing if `module = distribute_module(module,...)` and `distribute_module(module, ...)` are equivalent in my opinion.

Reply:
yes it's inplace modification, we don't need to return it if all people calling this API knows that it's a inplace  modification, but I think users won't care about too much details about how this function is implemented, what they care really is to call this function, and it returns a module that can be used for distributed training, I think this actually make a better UX, and it also aligns with `distribute_tensor` API offers.

Reply:
No problem. I felt a bit confused because I saw people using both `module = distribute_module(module,...)` and `distribute_module(module, ...)` in different tests in PR. Thx for the explanation!